sdk/servicebus/service-bus/samples/typescript/src/advanced/sessionRoundRobin.ts

```diff
@@ -142,15 +142,24 @@ async function roundRobinThroughAvailableSessions(): Promise<void> {
   const serviceBusClient = ServiceBusClient.createFromConnectionString(serviceBusConnectionString);
   const queueClient = serviceBusClient.createQueueClient(queueName);
 
+  const receiverPromises = [];
+
   for (let i = 0; i < maxSessionsToProcessSimultaneously; ++i) {
-    (async () => {
-      while (!abortController.signal.aborted) {
-        await receiveFromNextSession(queueClient);
-      }
-    })();
+    receiverPromises.push(
+      (async () => {
+        while (!abortController.signal.aborted) {
+          await receiveFromNextSession(queueClient);
+        }
+      })()
+    );
   }
 
   console.log(`Listening for available sessions...`);
+  await Promise.all(receiverPromises);
```

Comment:
The promises would never resolve because of the while loop right?
The only way out is if any of the promises get rejected in which case we dont reach the next line to close the queue client...

Reply:
Most of the time this won't do anything.

However if you want to use the abort signal to abort things you need to have some sort of trigger to say "close the service bus client" else your process won't ever exit :)

Reply:
Looks like what I missed was that the promise returned by `receiveFromNextSession` is never rejected because any error from the inside is handled by calling the user's error handler and we resolve the promise gracefully. In which case, yes we will be able to get to the part where we clean up the clients

Should we do a similar update to the sample in Track 2?

Reply:
Yes, absolutely.

Reply:
Yes! I updated the original issue to add some little checkboxes so I do the track 2 update as well.